lib/chef/provider/ifconfig.rb

```diff
@@ -59,15 +59,24 @@ def load_current_resource
 
         @status = shell_out("ifconfig")
         @status.stdout.each_line do |line|
-          if !line[0..9].strip.empty?
-            @int_name = line[0..9].strip
-            @interfaces[@int_name] = { "hwaddr" => (line =~ /(HWaddr)/ ? ($') : "nil").strip.chomp }
+          addr_regex = /^(\w+):?(\d*):?\ .+$/
```

Comment:
probably more like `/^(\S+):?\s/` although i think you need the magic to make the \S+ non-greedy.